Formula/swi-prolog.rb

```diff
@@ -0,0 +1,93 @@
+class SwiProlog < Formula
+  homepage "http://www.swi-prolog.org/"
+  url "http://www.swi-prolog.org/download/stable/src/swipl-7.2.3.tar.gz"
+  sha256 "43657d51b7c5887bc2d2bced50a9822b86a08a6841399b8e76ee877f51d646b5"
+
+  bottle do
+    revision 1
+    sha256 "d3d9dab5bb9f9ca0872ed6ecdff62fa09033775c3cb9f4f7b2a7f68f19c7a21a" => :el_capitan
+    sha256 "b6911b7573958dbc4ea8dbae03c06080239ce990e62a106ca03cdb6a9399d05d" => :yosemite
+    sha256 "193d6fc6f928dffda0826a1d01d04cfa6635764f4c343b0f14493a48455ebbe7" => :mavericks
+  end
+
+  devel do
+    url "http://www.swi-prolog.org/download/devel/src/swipl-7.3.19.tar.gz"
+    sha256 "693cfeac8f31d050ddcec27e550586359e98a64dc820398997d0ce6b8f564354"
+  end
+
+  head do
+    url "https://github.com/SWI-Prolog/swipl-devel.git"
+
+    depends_on "autoconf" => :build
+  end
+
+  option "with-lite", "Disable all packages"
+  option "with-jpl", "Enable JPL (Java Prolog Bridge)"
+  option "with-xpce", "Enable XPCE (Prolog Native GUI Library)"
+
+  deprecated_option "lite" => "with-lite"
+
+  depends_on "pkg-config" => :build
+  depends_on "readline"
+  depends_on "gmp"
+  depends_on "openssl"
+  depends_on "libarchive" => :optional
+
+  if build.with? "xpce"
+    depends_on :x11
+    depends_on "jpeg"
+  end
+
+  # 10.5 versions of these are too old
+  if MacOS.version <= :leopard
```

Comment:
Unless it makes life easier for @mistydemeo I'm not fussed about retaining these lines, given core formulae don't support anything newer than 10.6 and this conditional will consequently never be fulfilled.